src/main/java/com/github/maven_nar/Msvc.java

```diff
@@ -164,6 +170,12 @@ public void configureLinker(final LinkerDef linker) throws MojoExecutionExceptio
         for (File sdkLib : sdkLibs)
           linker.addLibraryDirectory(sdkLib, sdkArch);
 
+      final String envLib = System.getenv("LIB");
```

Comment:
This solution of hoisting in any global setting and not recording the configuration leads to variations in the build across build environments, which is undesirable and to be avoided where possible.
Scenarios of note:
 * build multi arch (x86/x64/arm) from the same command line
 * build multi linker (VC8/9/10/11/12/14/15) from the same command line
 * build server builds where you cannot (or it's difficult to) change the build environment before calling mvn, and where one environment might not be enough
 * vagaries between build hosts leading to incorrect headers/libs being included.

I've started a branch for VS 2017 changes, it was able to build 32 and 64 regardless of env
https://github.com/maven-nar/nar-maven-plugin/tree/feature/MSVC-VS2017

Includes refactoring and finding the appropriate target architecture specific libs as well as app paths.
I will be testing the plugin on a larger build as I resolve other codebase upgrade issues to VC15 with it, see if any other issues come out.  
Would be happy to hear any inputs on parts still needing to be resolved, or if something would then be broken for your build.